superset/assets/src/dashboard/reducers/getInitialState.js

```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
 
   const dashboard = { ...bootstrapData.dashboard_data };
   let filters = {};
-  try {
-    // allow request parameter overwrite dashboard metadata
-    filters = JSON.parse(
-      getParam('preselect_filters') || dashboard.metadata.default_filters,
-    );
-  } catch (e) {
-    //
+
+  const getFiltersFromFilterConfig = (filter_configs) => {
+    const filters = {};
+    filter_configs.forEach( filter => {
+      const defaultValue = filter.hasOwnProperty("defaultValue")?filter["defaultValue"]:"";
+      var valArr = defaultValue!==""?defaultValue.split(";"):[]; 
+      if(valArr.length > 0){
+        filters[filter["column"]] = valArr;
+      }
+    })
+    return filters;
   }
```

Comment:
1.correct code formatting 
2. we should trim string before add empty check or better way to validate empty string
3. add column prop check

Reply:
1. done
2. As we discussed not needed...component behaviour is like this
3. Not required as per my view column is compulsory filed in chart

Reply:
k